pkg/gofr/container/mock_container.go

```diff
@@ -32,8 +33,16 @@ func NewMockContainer(t *testing.T) (*Container, Mocks) {
 
 	ctrl := gomock.NewController(t)
 
-	sqlMock := NewMockDB(ctrl)
-	container.SQL = sqlMock
+	mockDB, sqlMock, _ := sql.NewSQLMocks(t)
+	// initialisation of expectations
+	e := expectedQuery{}
+
+	sql2 := &mockSQL{sqlMock, &e}
```

Comment:
`sql2` name is not very clear. Maybe we can name it to `sqlMockWrapper`, `testSQLMock` or anything more relevant?